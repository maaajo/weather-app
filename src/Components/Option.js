import React from 'react';

class Option extends React.Component {
  render() {
    const {
      id = '',
      name = '',
      bold = false,
      textContent = '',
      handleClick = () => {}
    } = this.props;

    return (
      <button
        id={id ? id : null}
        name={id && name ? `${name}-${id}` : null}
        type="button"
        className={`tracking-wider hover:underline ${bold ? 'font-bold' : ''}`}
        onClick={handleClick}
      >
        {textContent}
      </button>
    );
  }
}

export default Option;
