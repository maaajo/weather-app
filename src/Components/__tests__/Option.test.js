import React from 'react';
import { shallow, configure } from 'enzyme';
import Option from '../Option';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('prop test', () => {
  const mockFunction = jest.fn();
  const component = shallow(
    <Option
      name="language"
      id="en"
      textContent="en"
      handleClick={mockFunction}
      bold={true}
    />
  );
  const button = component.find('button');

  test('renders properly', () => {
    expect(button.debug()).toMatchSnapshot();
  });

  test('name created when id and name passed', () => {
    expect(button.prop('name')).toEqual('language-en');
  });

  test('id created when prop passed', () => {
    expect(button.prop('id')).toEqual('en');
  });
  test('text content should equal to passed prop', () => {
    expect(button.text()).toEqual('en');
  });
  test('triggers on click prop function when clicked', () => {
    button.simulate('click');
    expect(mockFunction).toHaveBeenCalledTimes(1);
  });
  test('should have class font-bold when bold is true', () => {
    expect(button.hasClass('font-bold')).toEqual(true);
  });
});
