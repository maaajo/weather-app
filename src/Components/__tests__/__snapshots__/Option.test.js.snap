// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prop test renders properly 1`] = `
"<button id=\\"en\\" name=\\"language-en\\" type=\\"button\\" className=\\"tracking-wider hover:underline font-bold\\" onClick={[Function: mockConstructor]}>
  en
</button>"
`;
